module/product/src/Infrastructure/Persistence/Projector/DbalProductValueChangedEventProjector.php

```diff
@@ -68,7 +68,9 @@ private function insertValue(string $productId, string $attributeId, ValueInterf
             $this->insert($productId, $attributeId, $phrase);
         } elseif ($value instanceof StringCollectionValue) {
             foreach ($value->getValue() as $language => $phrase) {
-                $this->insert($productId, $attributeId, $phrase, $language);
+                if ($phrase) {
```

Comment:
This is not a solution, this only ensures that an empty value is not saved in this projector.
Empty value will remain in the domain object and will appear in other projectors based on this event.